### PR TITLE
Feature/dog profile navigation

### PR DESCRIPTION
**Pull request by philyum**

### CODE REVIEW

**Comment by imann24** [comment]
@philyum: Please resolve merge conflicts w/ master

**Comment by imann24** [comment]
Getting a null ref when I click on scouting slots in the living room

**Comment by imann24** [comment]
I believe (leaving to @gracebarsny to confirm, Dog Profiles should be navigable on both **(a)** the "All Dogs" version of the "Dog Browser" and **(b)** on the shelter screen. 

I am only observing implementation on **(b)**, and I only see the clickable arrows as functional, I don't seem to be able to swipe to change pages. There is already a `UIAction` setup to handle this functionality which you can add in: https://github.com/PickupPup/Pickup-Pup/blob/master/Assets/Scripts/UI/Core/EventSystem/UIPageSwitchAction.cs

File: Assets/Scripts/UI/PickupPup/Core/PPUIController.cs [hunk 1]
```diff
@@ -89,12 +90,13 @@ void handlePPDogEvent(PPEvent gameEvent, Dog dog)
...
+			selectedSlot = dog.OccupiedSlot;
+			handleDogSlotClicked(dog);
 		}
 	}
 
-	void handleDogSlotClicked(Dog dog)
+	protected virtual void handleDogSlotClicked(Dog dog)
 	{
         // TODO: Insert universal dog slot handle code here
         if (dogProfileObject)
```
Comment (imann24) [inline Assets/Scripts/UI/PickupPup/Core/PPUIController.cs]: Remove space between `if` and paren

File: Assets/Scripts/UI/PickupPup/DogProfile/DogProfile.cs [hunk 2]
```diff
@@ -44,11 +39,12 @@ protected override void setReferences()
     {
         base.setReferences();
 
-        descriptionText = new Text[descriptionFields.Length];
-        for(int i = 0; i < descriptionText.Length; i++)
-        {
-            descriptionText[i] = descriptionFields[i].GetComponentInChildren<Text>();
-        }
+		if (descriptionFields != null) {
```
Comment (imann24) [inline Assets/Scripts/UI/PickupPup/DogProfile/DogProfile.cs]: - Remove space between `if` and paren
- opening bracket on newline

File: Assets/Scripts/UI/PickupPup/DogProfile/DogProfile.cs [hunk 3]
```diff
@@ -44,11 +39,12 @@ protected override void setReferences()
...
+			descriptionText = new Text[descriptionFields.Length];
+			for (int i = 0; i < descriptionText.Length; i++) {
```
Comment (imann24) [inline Assets/Scripts/UI/PickupPup/DogProfile/DogProfile.cs]: - Remove space between `for` and paren

Comment (imann24) [inline Assets/Scripts/UI/PickupPup/DogProfile/DogProfile.cs]: - opening bracket on newline

File: Assets/Scripts/UI/PickupPup/DogProfile/DogProfile.cs [hunk 4]
```diff
@@ -44,11 +39,12 @@ protected override void setReferences()
...
+				descriptionText [i] = descriptionFields [i].GetComponentInChildren<Text> ();
```
Comment (imann24) [inline Assets/Scripts/UI/PickupPup/DogProfile/DogProfile.cs]: Remove space between func name and paren:
`GetComponentInChildren<Text>();`

File: Assets/Scripts/UI/PickupPup/DogProfile/DogProfile.cs [hunk 5]
```diff
@@ -69,18 +65,16 @@ public virtual void SetProfile(Dog dog)
...
-            {
-                descriptionFields[i].Show();
-                descriptionText[i].text = dogInfo.Descriptions[i];
-            }
-            else
-            {
-                descriptionFields[i].Hide();
-            }
-        }
+		if (descriptionText != null) {
```
Comment (imann24) [inline Assets/Scripts/UI/PickupPup/DogProfile/DogProfile.cs]: Paren spacing and bracket newline note for this enter block (also applies to `else` statements)

File: Assets/Scripts/UI/PickupPup/DogProfile/DogProfileButtonController.cs [hunk 6]
```diff
@@ -0,0 +1,109 @@
...
+		dogsList = dogFactory.CreateGroupList(new List<DogDescriptor>(dogInfos));
+	}
+
+    void switchToProfile(int index)
+    {
+        checkReferences();
+        currentProfileIndex = index;
+        checkCurrentIndex();
+        parentWindow.SetProfile(dogsList[currentProfileIndex]);
+		if (OnSwitchProfile != null)
```
Comment (imann24) [inline Assets/Scripts/UI/PickupPup/DogProfile/DogProfileButtonController.cs]: - `if` statements should always including brackets in our codebase
- no space between paren and if

File: Assets/Scripts/UI/PickupPup/DogProfile/DogProfileButtonController.cs [hunk 7]
```diff
@@ -0,0 +1,109 @@
+﻿/*
+ * Author: Grace Barrett-Snyder
```
Comment (imann24) [inline Assets/Scripts/UI/PickupPup/DogProfile/DogProfileButtonController.cs]: Add your name in the header to any scripts you worked on

File: Assets/Scripts/UI/PickupPup/DogProfile/DogProfileButtonController.cs [hunk 8]
```diff
@@ -0,0 +1,109 @@
...
+        {
+            currentProfileIndex = dogsList.Count - 1;
+        }
+        else if(currentProfileIndex >= dogsList.Count)
+        {
+            currentProfileIndex = currentProfileIndex = 0;
+        }
+    }
+
+	public void SetCurrentIndex(int index) {
```
Comment (imann24) [inline Assets/Scripts/UI/PickupPup/DogProfile/DogProfileButtonController.cs]: Bracket on newline

File: Assets/Scripts/UI/PickupPup/ShelterScreen/PPShelterUIController.cs [hunk 9]
```diff
@@ -17,14 +20,31 @@ protected override void setReferences()
         promptID = PromptID.ShelterPrompt;
         base.setReferences();
         availableDogPortraits = GetComponentsInChildren<DogSlot>();
+		if (dogProfileObject != null)
```
Comment (imann24) [inline Assets/Scripts/UI/PickupPup/ShelterScreen/PPShelterUIController.cs]: Spacing with `if` statements

File: Assets/Scripts/UI/PickupPup/ShelterScreen/PPShelterUIController.cs [hunk 10]
```diff
@@ -17,14 +20,31 @@ protected override void setReferences()
...
 
     protected override void fetchReferences()
     {
         base.fetchReferences();
         database = DogDatabase.GetInstance;
         EventController.Event(PPEvent.LoadShelter);
-        populateAvailableDogs(database);
+        var dogs = populateAvailableDogs(database);
+		dogProfileButtonController.Init (dogShelterProfile, dogs);
+		dogProfileButtonController.OnSwitchProfile += (index) => {
```
Comment (imann24) [inline Assets/Scripts/UI/PickupPup/ShelterScreen/PPShelterUIController.cs]: bracket on newline

File: Assets/Scripts/UI/PickupPup/ShelterScreen/PPShelterUIController.cs [hunk 11]
```diff
@@ -17,14 +20,31 @@ protected override void setReferences()
...
     }
 
     protected override void fetchReferences()
     {
         base.fetchReferences();
         database = DogDatabase.GetInstance;
         EventController.Event(PPEvent.LoadShelter);
-        populateAvailableDogs(database);
+        var dogs = populateAvailableDogs(database);
+		dogProfileButtonController.Init (dogShelterProfile, dogs);
```
Comment (imann24) [inline Assets/Scripts/UI/PickupPup/ShelterScreen/PPShelterUIController.cs]: No space between func name and paren

File: Assets/Scripts/UI/PickupPup/ShelterScreen/PPShelterUIController.cs [hunk 12]
```diff
@@ -17,14 +20,31 @@ protected override void setReferences()
...
+		}
     }
 
     protected override void fetchReferences()
     {
         base.fetchReferences();
         database = DogDatabase.GetInstance;
         EventController.Event(PPEvent.LoadShelter);
-        populateAvailableDogs(database);
+        var dogs = populateAvailableDogs(database);
```
Comment (imann24) [inline Assets/Scripts/UI/PickupPup/ShelterScreen/PPShelterUIController.cs]: Think we should avoid use of `var` when possible. Prefer explicitly indicating type

**Comment by gracebarsny** [comment]
Just want to confirm Isaiah is right. The dog profile needs to work in both places.

**Comment by imann24** [comment]
@philyum please resolve MCs (merge conflicts)

**Comment by imann24** [comment]
@philyum Please review Technical Design Document and adhere to style conventions

**Pull request merged**